watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -0,0 +1,329 @@
+            * pyunits.convert(
+                m.fs.retentate1.flow_mass_comp[0, "dye"],
+                to_units=pyunits.kg / m.fs.costing.base_period,
+            )
+        ),
+        doc="Dollar value of dye in the retentate priced by mass",
+    )
+
+    m.fs.costing.annual_disposal_cost = Expression(
+        expr=(
+            m.fs.costing.utilization_factor
+            * (
+                m.fs.costing.waste_disposal_cost
+                * pyunits.convert(
+                    m.fs.permeate1.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.costing.base_period,
+                )
+            )
+        ),
+        doc="Annual cost of waste disposal",
+    )
+
+    m.fs.costing.value_dye_recovery_flow = Expression(
+        expr=(
+            m.fs.costing.utilization_factor
+            * (
+                m.fs.costing.dye_retentate_cost
+                * pyunits.convert(
+                    m.fs.retentate1.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.costing.base_period,
+                )
+            )
+        ),
+        doc="Annual value of dye-retentate recovery",
+    )
+    m.fs.costing.LCOW_dye_recovered = Expression(
+        expr=(
+            m.fs.costing.total_capital_cost * m.fs.costing.capital_recovery_factor
+            + m.fs.costing.total_operating_cost
+            + m.fs.costing.utilization_factor
+            * (
+                m.fs.costing.waste_disposal_cost
+                * pyunits.convert(
+                    m.fs.permeate1.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.costing.base_period,
+                )
+                - m.fs.costing.dye_retentate_cost
+                * pyunits.convert(
+                    m.fs.retentate1.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.costing.base_period,
+                )
+            )
+        )
+        / (
+            m.fs.costing.utilization_factor
+            * pyunits.convert(
+                m.fs.feed.flow_vol[0],
+                to_units=pyunits.m**3 / m.fs.costing.base_period,
+            )
+        ),
+        doc="Levelized cost of water treated with dye recovery through NF",
+    )
+
+    m.fs.costing.LCOT_dye_mass = Expression(
+        expr=(
+            m.fs.costing.total_capital_cost * m.fs.costing.capital_recovery_factor
+            + m.fs.costing.total_operating_cost
+            + m.fs.costing.utilization_factor
+            * m.fs.costing.waste_disposal_cost
+            * pyunits.convert(
+                m.fs.permeate1.properties[0].flow_vol,
+                to_units=pyunits.m**3 / m.fs.costing.base_period,
+            )
+            - m.fs.costing.value_dye_recovery_mass
+        )
+        / m.fs.costing.dye_recovery_mass,
+        doc="Levelized cost of treatment on a basis of dye recovered",
+    )
+
+
+def display_results(m):
+    unit_list = ["feed", "permeate1", "retentate1", "nanofiltration"]
```

Comment:
`permeate1` and `retentate1` are streams, not units, correct? Should this line instead be 

    unit_list = ["feed", "P1", "nanofiltration"]

Or have I misunderstood something?

Reply:
I see, thanks for the catch! In that case, I believe I should remove the feed stream from the display as well.

Reply:
Feed is modeled with a `FeedZO` block, so I believe it is appropriate to include it in the list here. Some of our other flowsheets have done so (e.g., biomembrane_filtration and metab).